src/mbgl/map/source.hpp

```diff
@@ -46,6 +47,7 @@ class SourceInfo : private util::noncopyable {
     std::string attribution;
     std::array<float, 3> center = {{0, 0, 0}};
     std::array<float, 4> bounds = {{-180, -90, 180, 90}};
+    std::string source_id = "";
```

Comment:
It's a way of finding the single (multi-layer) shape annotations source later in the tile update step. It's not the only vector source, it doesn't have a `url`, it doesn't have a `tiles` template, and it doesn't have `attribution`, so there is no other way to get a pointer to it later.